aea/cli/init.py

```diff
@@ -24,49 +24,64 @@
 from aea import __version__
 from aea.cli.login import do_login
 from aea.cli.register import do_register
-from aea.cli.registry.settings import AUTH_TOKEN_KEY
+from aea.cli.registry.settings import DEFAULT_REGISTRY_CONFIG, REGISTRY_CONFIG_KEY
 from aea.cli.registry.utils import check_is_author_logged_in, is_auth_token_present
 from aea.cli.utils.config import get_or_create_cli_config, update_cli_config
 from aea.cli.utils.constants import AEA_LOGO, AUTHOR_KEY
 from aea.cli.utils.context import Context
 from aea.cli.utils.decorators import pass_ctx
-from aea.cli.utils.package_utils import validate_author_name
+from aea.cli.utils.package_utils import validate_author_name, validate_registry_type
 
 
 @click.command()
 @click.option("--author", type=str, required=False)
+@click.option("--default-registry", type=str, required=False)
 @click.option("--reset", is_flag=True, help="To reset the initialization.")
 @click.option("--local", is_flag=True, help="For init AEA locally.")
 @click.option("--no-subscribe", is_flag=True, help="For developers subscription.")
 @pass_ctx
 def init(  # pylint: disable=unused-argument
-    ctx: Context, author: str, reset: bool, local: bool, no_subscribe: bool
+    ctx: Context,
+    author: str,
+    reset: bool,
+    local: bool,
+    no_subscribe: bool,
+    default_registry: str,
 ) -> None:
     """Initialize your AEA configurations."""
-    do_init(author, reset, not local, no_subscribe)
+    do_init(author, reset, not local, no_subscribe, default_registry)
 
 
-def do_init(author: str, reset: bool, registry: bool, no_subscribe: bool) -> None:
+def do_init(
+    author: str, reset: bool, registry: bool, no_subscribe: bool, default_registry: str
+) -> None:
     """
     Initialize your AEA configurations.
 
     :param author: str author username.
     :param reset: True, if resetting the author name
     :param registry: True, if registry is used
     :param no_subscribe: bool flag for developers subscription skip on register.
+    :param default_registry: default registry type.
     """
     config = get_or_create_cli_config()
     if reset or config.get(AUTHOR_KEY, None) is None:
         author = validate_author_name(author)
+        registry_type = validate_registry_type(default_registry)
         if registry:
             _registry_init(username=author, no_subscribe=no_subscribe)
 
+        registry_config = DEFAULT_REGISTRY_CONFIG.copy()
+        registry_config.update({"default": registry_type})
+
+        update_cli_config({REGISTRY_CONFIG_KEY: registry_config})
```

Comment:
I've added a config field for registry config with the following format
```yaml
registry_config:
  default: ipfs
  settings:
    http:
      auth_token: null
      registry_api_url: null
    ipfs: {}
```
Now we can define which back-end(http, ipfs) to use when running `aea ini`.